### PR TITLE
Changes from ukub mission...

### LINKED ISSUE #11: SDIO Support

### PR DESCRIPTION
**Pull request by rplauche**
* Overrode HAL tick functions to use FreeRTOS tick
* Tightening I2C timing and changes to support clock stretching
* Switched UART driver from the CubeF4 USART interface to the UART interface (Fixes UART RX)
* Implemented SDIO driver (Closes #11)

### CODE REVIEW

File: source/sdio.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        return SDIO_INIT_ERROR;
+    }
+
+    if (sd_handle.Instance != NULL)
+    {
+        return SDIO_OK;
+    }
+    HAL_SD_CardInfoTypedef card_info;
+
+    sdio_msp_init();
```
Comment (marshall) [inline source/sdio.c]: Should this be `sdio_stm_init`, or does the `msp` stand for something else? :)

File: source/sdio.c [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        return SDIO_OK;
+    }
+    return SDIO_ERROR;
+}
+
+static void sdio_msp_init(void)
+{
+    GPIO_InitTypeDef GPIO_Init_Structure;
+
+	SET_BIT(RCC->AHB1ENR,
```
Comment (marshall) [inline source/sdio.c]: nit: some kind of code spacing inconsistency again. we should really look into a code linter :)

File: source/uart.c [hunk 3]
```diff
@@ -95,61 +96,67 @@ void kprv_uart_dev_init(KUARTNum uart)
...
         default:
-            u.Init.WordLength = USART_WORDLENGTH_8B; break;
+            u.Init.WordLength = UART_WORDLENGTH_8B; break;
     }
 
     switch (k_uart->conf.stop_bits) {
-        case K_STOP_BITS_1_5:
-            u.Init.StopBits = USART_STOPBITS_1_5; break;
+        // case K_STOP_BITS_1_5:
+        //     u.Init.StopBits = UART_STOPBITS_1_5; break;
```
Comment (marshall) [inline source/uart.c]: nit: remove commented code

File: source/uart.c [hunk 4]
```diff
@@ -182,22 +192,51 @@ static inline void uart_irq_handler(KUARTNum uart)
...
+        ((void)tmpreg);
+    }
+
+    if (__GET_FLAG(dev, USART_SR_FE))
+    {
+        // clear out the data register on framing error
+        volatile uint32_t tmpreg = dev->SR;
+        tmpreg = dev->DR;
+        ((void)tmpreg);
+    }
```
Comment (marshall) [inline source/uart.c]: lots of repetitive code here, do we not already have a macro for this logic somewhere?

**Comment by marshall** [comment]
Looks mostly ready to merge, see comments and then I think it's ready to go

File: source/sdio.c (hunk 1 above)
Comment (rplauche) [inline source/sdio.c]: This is modeled after the STM32CubeF4 HAL's naming convention, where the low level hardware init functions are called `HAL_*_MspInit`. Msp appears to stand for `MCU Specific Package`.

**Pull request merged**